internal/dag/gatewayapi_processor.go

```diff
@@ -433,3 +457,18 @@ func (p *GatewayAPIProcessor) cluster(headerPolicy *HeadersPolicy, service *Serv
 		RequestHeadersPolicy: headerPolicy,
 	}
 }
+
+func pathMatchTypePtr(pmt gatewayapi_v1alpha1.PathMatchType) *gatewayapi_v1alpha1.PathMatchType {
+	return &pmt
+}
+
+func headerMatchTypePtr(hmt gatewayapi_v1alpha1.HeaderMatchType) *gatewayapi_v1alpha1.HeaderMatchType {
+	return &hmt
+}
+
+func stringOrDefaultFromPtr(s *string, def string) string {
```

Comment:
I think this can be replaced by using `pointer.StringDeref` from the k8s util package

Reply:
👍